File: Code/HWQATool/HWQATool/Controllers/SubTaskController.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Entity;
using System.Linq;
using System.Net;
using System.Web;
using System.Web.Mvc;
using HWQATool.Models;

namespace HWQATool.Controllers
{
    public class SubTaskController : Controller
    {
        private HWQAToolContext db = new HWQAToolContext();

        // GET: /SubTask/
        public ActionResult Index()
        {
            var subtasks = db.SubTasks.Include(s => s.Task);
            return View(subtasks.ToList());
        }

        // GET: /SubTask/Details/5
        public ActionResult Details(int? id)
        {
            if (id == null)
            {
                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
            }
            SubTask subtask = db.SubTasks.Find(id);
            if (subtask == null)
            {
                return HttpNotFound();
            }
            return View(subtask);
        }

        // GET: /SubTask/Create
        public ActionResult Create()
        {
            ViewBag.TaskId = new SelectList(db.Functions, "Id", "Name");
            return View();
        }

        // POST: /SubTask/Create
        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult Create([Bind(Include="Id,Name,TaskId,Version,LastModifiedAt,LastModifiedBy")] SubTask subtask)
        {
            if (ModelState.IsValid)
            {
                db.SubTasks.Add(subtask);
                db.SaveChanges();
                return RedirectToAction("Index");
            }

            ViewBag.TaskId = new SelectList(db.Functions, "Id", "Name", subtask.TaskId);
            return View(subtask);
        }

        // GET: /SubTask/Edit/5
        public ActionResult Edit(int? id)
        {
            if (id == null)
            {
                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
            }
            SubTask subtask = db.SubTasks.Find(id);
            if (subtask == null)
            {
                return HttpNotFound();
            }
            ViewBag.TaskId = new SelectList(db.Functions, "Id", "Name", subtask.TaskId);
            return View(subtask);
        }

        // POST: /SubTask/Edit/5
        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult Edit([Bind(Include="Id,Name,TaskId,Version,LastModifiedAt,LastModifiedBy")] SubTask subtask)
        {
            if (ModelState.IsValid)
            {
                db.Entry(subtask).State = EntityState.Modified;
                db.SaveChanges();
                return RedirectToAction("Index");
            }
            ViewBag.TaskId = new SelectList(db.Functions, "Id", "Name", subtask.TaskId);
            return View(subtask);
        }

        // GET: /SubTask/Delete/5
        public ActionResult Delete(int? id)
        {
            if (id == null)
            {
                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
            }
            SubTask subtask = db.SubTasks.Find(id);
            if (subtask == null)
            {
                return HttpNotFound();
            }
            return View(subtask);
        }

        // POST: /SubTask/Delete/5
        [HttpPost, ActionName("Delete")]
        [ValidateAntiForgeryToken]
        public ActionResult DeleteConfirmed(int id)
        {
            SubTask subtask = db.SubTasks.Find(id);
            db.SubTasks.Remove(subtask);
            db.SaveChanges();
            return RedirectToAction("Index");
        }

        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                db.Dispose();
            }
            base.Dispose(disposing);
        }
    }
}

File: Sample/Sample/UI/controller/subtask.controller.js
app.controller("subtaskController", function ($scope,subtaskService, $http) {

    subtaskService.get(function (data) {
        $scope.subtasks = data;
    });
    debugger;
  

    $scope.defaults = function () {
        $scope.subtask = {
            id: 0,
            name: "",
            taskId: 0

        }

    }

    $scope.addSubTask = function () {
        debugger;
        if ($scope.subtask.id != 0) {
            $http.put("http://localhost:2864/api/SubTasks" + $scope.subtask.id, $scope.subtask).then(function (response) {
                debugger;
            }, function (error) {

            });
        }
        else {
            $http.post("http://localhost:2864/api/SubTasks", $scope.subtask).then(function (response) {
                $scope.subtasks.push($scope.subtask);
            }, function (error) {

            });
        }
    }

    $scope.deleteSubTask = function (index) {
        debugger;
        $scope.subtask = $scope.subtasks[index];
        $http.delete("http://localhost:2864/api/SubTasks/" + $scope.subtask.id).then(function (response) {
            debugger;
            $scope.subtasks.splice(index, 1);
        }, function (error) {
        });

    }



    $scope.editSubTask = function () {
        $scope.subtask = $scope.subtasks[index];

    }
}
);